pallets/fellowship/src/impls.rs

```diff
@@ -0,0 +1,79 @@
+use crate::traits::EnsureRole;
+use crate::*;
+use common_traits::MaybeConvert;
+use frame_support::{ensure, traits::Get};
+use sp_runtime::{
+    traits::{BadOrigin, Convert},
+    DispatchError, Percent,
+};
+use sp_std::vec::Vec;
+use orml_traits::MultiReservableCurrency;
+
+/// Ensure that a account is of a given role.
+/// Used in other pallets like an ensure origin.
+pub struct EnsureFellowshipRole<T>(T);
+impl<T: Config> EnsureRole<AccountIdOf<T>, Role> for EnsureFellowshipRole<T> {
+    type Success = ();
+
+    fn ensure_role(
+        acc: &AccountIdOf<T>,
+        role: Role,
+        rank: Option<Rank>,
+    ) -> Result<Self::Success, DispatchError> {
+        let (actual_role, actual_rank) = Roles::<T>::get(acc).ok_or(BadOrigin)?;
+        ensure!(actual_role == role, BadOrigin);
+        if let Some(r) = rank {
+            ensure!(r == actual_rank, BadOrigin);
+        }
+        Ok(())
+    }
+    fn ensure_role_in(
```

Comment:
cant we use one function

seems both are similar except one checks a vec? 

Are you concerned about the potential compute/block time?

Reply:
The idea behind this was so we can just use an `ensure_role_in` instead of doing multiple `ensure_role` 
As for compute time i guess its the developers responsibility to ensure the input isn't from the user and isn't over bounds.

Reply:
Got it, thanks. LGTM!